src/mono/mono/mini/interp/jiterpreter.c

```diff
@@ -884,7 +884,8 @@ jiterp_insert_entry_points (void *_imethod, void *_td)
 		// Increase the instruction counter. If we inserted an entry point at the top of this bb,
 		//  the new instruction counter will be the number of instructions in the block, so if
 		//  it's big enough we'll be able to insert another entry point right away.
-		instruction_count += bb->in_count;
```

Comment:
For accurate count computation it would be a good idea to check if `MINT_IS_EMIT_NOP`